lib/concurrent/executor/executor.rb

```diff
@@ -59,6 +59,38 @@ def handle_fallback(*args)
     def serialized?
       false
     end
+
+    def auto_terminate?
+      !! @auto_terminate
+    end
+
+    protected
+
+    def enable_at_exit_handler!(opts = {})
+      if opts.fetch(:stop_on_exit, true)
+        @auto_terminate = true
+        if RUBY_PLATFORM == 'java'
+          create_java_at_exit_handler!(self)
+        else
+          create_ruby_at_exit_handler!(self.object_id)
+        end
+      end
+    end
+
+    def create_ruby_at_exit_handler!(id)
+      at_exit do
+        if Concurrent.auto_terminate_all_executors?
+          this = ObjectSpace._id2ref(id)
+          this.kill if this
+        end
+      end
+    end
+
+    def create_java_at_exit_handler!(this)
+      at_exit do
+        this.kill if Concurrent.auto_terminate_all_executors?
```

Comment:
Probably, but I don't know is that's avoidable. But I also can't image an application creating more than a handful of thread pools. I would expect that most would simply use the global thread pools and those are going to live until the application exits. The original implementation used `self` within the `at_exit` block but I don't think that was safe.